pojo-generator-core/src/test/java/uk/gov/justice/generation/pojo/plugin/factory/PluginInstantiatorTest.java

```diff
@@ -0,0 +1,225 @@
+package uk.gov.justice.generation.pojo.plugin.factory;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.when;
+
+import uk.gov.justice.generation.pojo.dom.ClassDefinition;
+import uk.gov.justice.generation.pojo.plugin.FactoryMethod;
+import uk.gov.justice.generation.pojo.plugin.PluginProviderException;
+import uk.gov.justice.generation.pojo.plugin.classmodifying.ClassModifyingPlugin;
+import uk.gov.justice.generation.pojo.plugin.classmodifying.PluginContext;
+
+import java.lang.reflect.Method;
+
+import com.squareup.javapoet.TypeSpec;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@SuppressWarnings("unchecked")
+@RunWith(MockitoJUnitRunner.class)
+public class PluginInstantiatorTest {
+
+    @Mock
+    private Instantiator instantiator;
+
+    @InjectMocks
+    private PluginInstantiator pluginInstantiator;
+
+    @Test
+    public void shouldInstantiateAPluginUsingItsFactoryMethod() throws Exception {
+
+        final PluginWithFactoryMethod pluginWithFactoryMethod = PluginWithFactoryMethod.createNew();
+        final Class aClass = pluginWithFactoryMethod.getClass();
```

Comment:
aClass...I like the names in the other tests below

Reply:
fixed